examples/rust/custom_space_view/src/color_coordinates_space_view.rs

```diff
@@ -0,0 +1,248 @@
+
+/// Space view state for the custom space view.
+///
+/// This state is preserved between frames, but not across Viewer sessions.
+#[derive(Default)]
+pub struct ColorCoordinatesSpaceViewState {
+    // TODO(wumpf/jleibs): This should be part of the Blueprint so that it is serialized out.
+    //                      but right now there is no way of doing that.
+    mode: ColorCoordinatesMode,
+}
+
+impl SpaceViewState for ColorCoordinatesSpaceViewState {
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    fn as_any_mut(&mut self) -> &mut dyn std::any::Any {
+        self
+    }
+}
+
+#[derive(Default)]
+pub struct ColorCoordinatesSpaceView;
+
+impl SpaceViewClass for ColorCoordinatesSpaceView {
+    // State type as described above.
+    type State = ColorCoordinatesSpaceViewState;
+
+    // Scene context is shared between all scene parts.
+    // For this Space View we don't need any scene context.
+    type Context = ();
+
+    // Collection of scene parts that are needed to display a frame.
+    // We only have a single scene part, but it could be a collection of various.
+    type SceneParts = SceneColorCoordinates;
+
+    // Scene parts can have a common data object that they expose.
+    // For this Space View this is not needed.
+    type ScenePartData = ();
+
+    fn name(&self) -> SpaceViewClassName {
+        // Name and identifier of this Space View.
+        "Color Coordinates".into()
+    }
+
+    fn icon(&self) -> &'static re_ui::Icon {
+        &re_ui::icons::SPACE_VIEW_SCATTERPLOT
+    }
+
+    fn help_text(&self, _re_ui: &re_ui::ReUi, _state: &Self::State) -> egui::WidgetText {
+        "A demo space view that shows colors as coordinates on a 2D plane.".into()
+    }
+
+    fn preferred_tile_aspect_ratio(&self, _state: &Self::State) -> Option<f32> {
+        // Prefer a square tile if possible.
+        Some(1.0)
+    }
+
+    /// Additional UI displayed when the space view is selected.
+    ///
+    /// In this sample we show a combo box to select the color coordinates mode.
+    fn selection_ui(
+        &self,
+        _ctx: &mut ViewerContext<'_>,
+        ui: &mut egui::Ui,
+        state: &mut Self::State,
+        _space_origin: &EntityPath,
+        _space_view_id: SpaceViewId,
+    ) {
+        ui.horizontal(|ui| {
+            ui.label("Coordinates mode");
+            egui::ComboBox::from_id_source("color_coordinates_mode")
+                .selected_text(state.mode.to_string())
+                .show_ui(ui, |ui| {
+                    ui.style_mut().wrap = Some(false);
+                    ui.set_min_width(64.0);
+
+                    let mut selectable_value = |ui: &mut egui::Ui, e| {
+                        ui.selectable_value(&mut state.mode, e, e.to_string())
+                    };
+                    selectable_value(ui, ColorCoordinatesMode::Hs);
+                    selectable_value(ui, ColorCoordinatesMode::Hv);
+                    selectable_value(ui, ColorCoordinatesMode::Rg);
```

Comment:
alternative:

```suggestion
                    for mode in ColorCoordinatesMode::ALL {
                        ui.selectable_value(&mut state.mode, mode, mode.to_string())
                    }
```

And then adding a `ColorCoordinatesMode::ALL` member. In general this makes it less likely that one forgets to update the ui code when more enum variants are added